src/model.rs

```diff
@@ -1835,6 +1852,219 @@ mod tests {
             assert_eq!(1, stop_time.pickup_type);
             assert_eq!(0, stop_time.drop_off_type);
         }
+
+        #[test]
+        fn forbidden_drop_off_should_be_kept() {
+            // if restriction are explicitly set they should not be overriden
+            let model = transit_model_builder::ModelBuilder::default()
+                .vj("vj1", |vj| {
+                    vj.block_id("block_1")
+                        .st("SP1", "10:00:00", "10:01:00")
+                        .st_mut("SP2", "11:00:00", "11:01:00", |st| {
+                            st.pickup_type = 1;
+                            st.drop_off_type = 1;
+                        });
+                })
```

Comment:
It's a matter of preference. I think the canonical api modeling should be as simple as possible.
And I don't really see a case where you do not want to set the stop point and (departure or arrival) so this could lead to not correctly defined stoptimes.
I think we could even make the canonical path easier if we had an easy way to set the departure value to the arrival, but I don't see a nice way to do this :/

Reply:
I'm just worried that in some cases, canonical means something and some other cases, we discover that canonical could be something different (no departure and arrival for example, if you only care about the path and therefore the stop points)... Let's keep it like this for now and will improve over time.